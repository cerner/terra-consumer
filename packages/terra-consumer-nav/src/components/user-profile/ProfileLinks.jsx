import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './UserProfile.scss';

const cx = classNames.bind(styles);

const propTypes = {
  linkItems: PropTypes.arrayOf(PropTypes.shape({
    slug: PropTypes.string,
    nav_type: PropTypes.oneOf(['GROUPING', 'EXTERNAL', 'APPLICATION', 'MODAL']),
    text: PropTypes.string.isRequired,
    uri: PropTypes.string.isRequired,
    icon: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.element,
    ]),
  })),
};

const defaultProps = {
  linkItems: [],
};

const ProfileLinks = ({
  linkItems,
  ...customProps
}) => {
  const linkElements = linkItems.map((linkItem) => {
    const linkElement = (<a key={`${linkItem.text}`} href={linkItem.uri} onClick={() => window.location = linkItem.uri} className={cx('suppress-hyperlink')}>
      <div className={cx('link', 'divider-bottom')}>{linkItem.text}</div></a>);
    return linkElement;
  });

  return (
    <div {...customProps}>
      {linkElements}
    </div>
  );
};

ProfileLinks.propTypes = propTypes;
ProfileLinks.defaultProps = defaultProps;

export default ProfileLinks;
