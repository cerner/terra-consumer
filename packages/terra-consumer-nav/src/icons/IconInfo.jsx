/* eslint-disable */
import React from 'react';
import IconBase from 'terra-icon/lib/IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <path d="M24,1C11.317,1,1,11.318,1,24s10.317,23,23,23s23-10.318,23-23S36.683,1,24,1z M24,45C12.421,45,3,35.58,3,24S12.421,3,24,3s21,9.42,21,21S35.579,45,24,45z" />
      <circle cx="24" cy="36" r="2" />
      <path
        d="M24,11c-2.642,0-5.375,1.868-6.5,4.442c-0.222,0.506,0.01,1.096,0.516,1.317c0.508,0.222,1.097-0.01,1.316-0.516
        C20.141,14.395,22.147,13,24,13c2.663,0,5,2.336,5,5s-2.337,5-5,5c-0.553,0-1,0.448-1,1v5c0,0.552,0.447,1,1,1s1-0.448,1-1v-4.075
        c3.341-0.502,6-3.47,6-6.925C31,14.206,27.794,11,24,11z"
      />
    </IconBase>
  );
};

SvgIcon.displayName = "IconInfo";
SvgIcon.defaultProps = {"viewBox":"0 0 48 48","xmlns":"http://www.w3.org/2000/svg","data-name":"Layer 1"};

export default SvgIcon;
/* eslint-enable */
