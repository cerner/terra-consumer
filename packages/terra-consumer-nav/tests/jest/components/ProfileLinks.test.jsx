import React from 'react';
import ProfileLinks from '../../../src/components/user-profile/ProfileLinks';
import IconInfo from '../../../src/icons/IconInfo';

const testData = [
  {
    text: 'Account',
    icon: (<IconInfo />),
    uri: 'http://localhost:8080/',
  },
  {
    text: 'Notifications',
    icon: (<IconInfo />),
    uri: 'http://localhost:8080/',
  },
];

describe('Profile Link items', () => {
  // Snapshot Tests
  it('should render a list of profile link items', () => {
    const wrapper = shallow(<ProfileLinks linkItems={testData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
