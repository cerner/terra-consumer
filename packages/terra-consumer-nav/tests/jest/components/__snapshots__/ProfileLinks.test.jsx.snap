// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Link items should render a list of profile link items 1`] = `
<div>
  <ProfileLink
    icon={
      <IconOutlineQuestionMark
        height={16}
        viewBox="0 0 48 48"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    text="Account"
    url="http://localhost:8080/"
  />
  <ProfileLink
    icon={
      <IconOutlineQuestionMark
        height={16}
        viewBox="0 0 48 48"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    text="Notifications"
    url="http://localhost:8080/"
  />
</div>
`;
